src/traffic_server/InkAPI.cc

```diff
@@ -4489,6 +4489,56 @@ TSContScheduleEvery(TSCont contp, ink_hrtime every, TSThreadPool tp)
   return action;
 }
 
+TSAction
+TSContScheduleOn(TSCont contp, TSHRTime timeout, TSEventThread *event_thread)
+{
+  ink_release_assert(event_thread != nullptr);
+
+  sdk_assert(sdk_sanity_check_iocore_structure(contp) == TS_SUCCESS);
+
+  FORCE_PLUGIN_SCOPED_MUTEX(contp);
+
+  INKContInternal *i = (INKContInternal *)contp;
```

Comment:
No C-style casts!

Reply:
This is copied from `TSContSchedule`, should I change those as well?